src/env.mjs

```diff
@@ -6,6 +6,12 @@ import { z } from 'zod';
 const zNodeEnv = () =>
   z.enum(['development', 'test', 'production']).default('development');
 
+const zOptionalWithReplaceMe = () =>
+  z
+    .string()
+    .optional()
+    .transform((value) => (value === 'REPLACE ME' ? undefined : value));
```

Comment:
I don't think so. 
Here if you kept "REPLACE ME" in "prod" it will be replaced with `undefined` so at least you can check if the value exist or not.
I think it's better than keeping "REPLACE ME" that is a "valide" value and then initiate some code with a known wrong value.

Reply:
I added some extra verification for production only